app/controllers/post_actions_controller.rb

```diff
@@ -48,21 +36,25 @@ def create
   end
 
   def destroy
-    post_action = current_user.post_actions.find_by(post_id: params[:id].to_i, post_action_type_id: @post_action_type_id, deleted_at: nil)
-    raise Discourse::NotFound if post_action.blank?
-
-    guardian.ensure_can_delete!(post_action)
-
-    PostAction.remove_act(current_user, @post, post_action.post_action_type_id)
-
-    @post.reload
-    render_post_json(@post, _add_raw = false)
+    result = PostActionDestroyer.new(
+      current_user,
+      Post.find_by(id: params[:id].to_i),
+      @post_action_type_id
+    ).perform
+
+    if result.failed?
+      render_json_error(result)
+    else
+      render_post_json(result.post, _add_raw = false)
```

Comment:
I know it's not your change, but we should probably move the `add_raw` optional parameter to a named parameter?

Reply:
Sure it's a small thing so I'll add it.